collect/collect.go

```diff
@@ -179,6 +182,104 @@ func setHostName() error {
 	return nil
 }
 
+type agMetric struct {
+	metric string
+	ts     opentsdb.TagSet
+	values []float64
+}
+
+func AggregateMeta(metric string, desc string, rateType metadata.RateType, unit metadata.Unit) {
+	agStrings := []string{"avg", "count", "min", "median", "max", "95", "99"}
+	for _, ag := range agStrings {
+		metadata.AddMetricMeta(metric+"_"+ag, rateType, unit, desc)
+	}
+}
+
+func (am *agMetric) Process(now int64) {
+	var avg float64
+	for _, v := range am.values {
+		avg += v
+	}
+	avg /= float64(len(am.values))
+	extRoot := metricRoot + am.metric
+	tchan <- &opentsdb.DataPoint{
+		Metric:    extRoot + "_avg",
+		Timestamp: now,
+		Value:     avg,
+		Tags:      am.ts,
+	}
+	tchan <- &opentsdb.DataPoint{
+		Metric:    extRoot + "_count",
+		Timestamp: now,
+		Value:     len(am.values),
+		Tags:      am.ts,
+	}
+	tchan <- &opentsdb.DataPoint{
+		Metric:    extRoot + "_min",
+		Timestamp: now,
+		Value:     percentile(am.values, 0),
+		Tags:      am.ts,
+	}
+	tchan <- &opentsdb.DataPoint{
+		Metric:    extRoot + "_median",
+		Timestamp: now,
+		Value:     percentile(am.values, .5),
+		Tags:      am.ts,
+	}
+	tchan <- &opentsdb.DataPoint{
+		Metric:    extRoot + "_max",
+		Timestamp: now,
+		Value:     percentile(am.values, 1),
+		Tags:      am.ts,
+	}
+	tchan <- &opentsdb.DataPoint{
+		Metric:    extRoot + "_95",
+		Timestamp: now,
+		Value:     percentile(am.values, .95),
+		Tags:      am.ts,
+	}
+	tchan <- &opentsdb.DataPoint{
+		Metric:    extRoot + "_99",
+		Timestamp: now,
+		Value:     percentile(am.values, .99),
+		Tags:      am.ts,
+	}
+}
```

Comment:
Do we want to clear the points after we process? Roll over n points or a fixed time window?